packages/database/test/database.test.ts

```diff
@@ -76,6 +76,20 @@ describe('Database Tests', function() {
     expect(db.ref().toString()).to.equal('https://localhost/');
   });
 
+  it('Can read ns query param', function() {
+    var db = defaultApp.database('http://localhost:80?ns=foo');
+    expect(db).to.be.ok;
+    expect(db.repo_.repoInfo_.namespace).to.equal('foo');
+    expect(db.ref().toString()).to.equal('http://localhost:80/');
+  });
+
+  it('Only reads ns query param when subdomain not set', function() {
+    var db = defaultApp.database('http://bar.firebaseio.com?ns=foo');
```

Comment:
Does this code work for `http://bar.firebaseio.com/?ns=foo`? I assume our rock-solid URL scheme parsing will then put this in the `path` instead.

Reply:
fixed